tests/atomistics/structure/test_neighbors.py

```diff
@@ -254,12 +254,54 @@ def test_get_local_shells_ragged(self):
         self.assertEqual(
             np.sum(neigh.get_local_shells(cluster_by_distances=True, cluster_by_vecs=True)==-1), 12
         )
-        neigh.allow_ragged = True
+        neigh.mode = 'ragged'
         self.assertEqual(np.sum([len(s)==11 for s in neigh.shells]), 12)
         self.assertEqual(np.sum([len(s)==11 for s in neigh.get_local_shells(cluster_by_distances=True)]), 12)
         self.assertEqual(np.sum([len(s)==11 for s in neigh.get_local_shells(cluster_by_vecs=True)]), 12)
         self.assertEqual(np.sum([len(s)==11 for s in neigh.get_local_shells(cluster_by_distances=True, cluster_by_vecs=True)]), 12)
 
+    def test_get_shells_flattened(self):
+        structure = StructureFactory().ase.bulk('Al', cubic=True).repeat(2)
+        del structure[0]
+        r = structure.cell[0,0]*0.49
+        neigh = structure.get_neighbors(cutoff_radius=r, num_neighbors=None, mode='flattened')
+        self.assertEqual(len(np.unique(neigh.shells)), 1)
+        self.assertEqual(len(neigh.shells), 360)
```

Comment:
I saw failing tests on the discussion page, and I bet this style of test is the main culprit now that you're using `np.iinfo(np.int32).max`